### ISSUE TITLE
ASM problem since last update

### ISSUE THREAD

**Issue by caay2000** [mjedynak/pit-idea-plugin#15]
Since last update, i always have problems running pitest plugin, seems to be an asm library incompatibility.

```
14:58:10 PIT >> INFO : Calculated coverage in 2 seconds.
Exception in thread "main" java.lang.IncompatibleClassChangeError: class org.objectweb.asm.tree.ClassNode has interface org.objectweb.asm.ClassVisitor as super class
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.pitest.bytecode.analysis.ClassTree.fromBytes(ClassTree.java:29)
	at org.pitest.mutationtest.build.MutationSource.createMutations(MutationSource.java:61)
	at org.pitest.mutationtest.build.MutationTestBuilder$2.apply(MutationTestBuilder.java:100)
	at org.pitest.mutationtest.build.MutationTestBuilder$2.apply(MutationTestBuilder.java:97)
	at org.pitest.functional.FCollection.flatMapTo(FCollection.java:55)
	at org.pitest.functional.FCollection.flatMap(FCollection.java:65)
	at org.pitest.mutationtest.build.MutationTestBuilder.createMutationTestUnits(MutationTestBuilder.java:56)
	at org.pitest.mutationtest.tooling.MutationCoverage.buildMutationTests(MutationCoverage.java:279)
	at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:134)
	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:115)
	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:49)
	at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
	at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)
```

**Comment by mjedynak** [comment]
Please specify: 

- IntelliJ version

- JDK version

- classpath from test execution

**Comment by caay2000** [comment]
Hi,

Intellij **CE 2017.2.5**
JDK **1.8.131**
Classpath:
```
com.bskyb.services:cbs-transactional-comms:war:2.0.232-SNAPSHOT
antlr:antlr:jar:2.7.6:compile
aopalliance:aopalliance:jar:1.0:compile
apache-httpclient:commons-httpclient:jar:3.1:compile
asm:asm:jar:3.3.1:compile
bouncycastle:bcprov-jdk14:jar:140:runtime
c3p0:c3p0:jar:0.9.1.2:compile
cglib:cglib-nodep:jar:2.2:compile
cglib:cglib:jar:2.2.2:compile
ch.qos.logback:logback-classic:jar:1.0.12:compile
ch.qos.logback:logback-core:jar:1.0.12:compile
com.fasterxml:classmate:jar:1.1.0:compile
com.fasterxml.jackson.core:jackson-annotations:jar:2.8.8:compile
com.fasterxml.jackson.core:jackson-core:jar:2.8.8:compile
com.fasterxml.jackson.core:jackson-databind:jar:2.8.8:compile
com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.8.8:compile
com.filenet:javaapi:jar:3.5.1:compile
com.github.hazendaz:javabean-tester:jar:1.4.0:compile
com.google.code.findbugs:annotations:jar:2.0.0:compile
com.google.code.findbugs:jsr305:jar:1.3.9:compile
com.google.code.gson:gson:jar:1.7.2:compile
com.google.guava:guava:jar:19.0:compile
com.googlecode.ehcache-spring-annotations:ehcache-spring-annotations:jar:1.2.0:compile
com.googlecode.json-simple:json-simple:jar:1.1:compile
com.googlecode.lambdaj:lambdaj:jar:2.3.3:compile
com.googlecode.libphonenumber:libphonenumber:jar:7.5.2:compile
com.ibm.mq:mq-tx-lib:pom:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-commonservices:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-jms-internal:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-jms:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-matchspace:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-provider:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-ref:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-wmq-common:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-wmq-factories:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-wmq-v6:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:client-wmq:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:commonservices-j2se:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:commonservices:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:connector:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:dhbcore:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:headers:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:jmqi-local:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:jmqi-remote:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:jmqi-system:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:jmqi:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:jms-admin:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:mq-jms:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:mq:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:mqetclient:jar:7.0.1-3:compile
com.ibm.mq.tx.et-client:pcf:jar:7.0.1-3:compile
com.mangofactory:swagger-models:jar:0.9.4:compile
com.mangofactory:swagger-springmvc:jar:0.9.4:compile
com.neovisionaries:nv-i18n:jar:1.20:compile
com.netflix.archaius:archaius-core:jar:0.4.1:runtime
com.netflix.hystrix:hystrix-core:jar:1.4.14:compile
com.netflix.hystrix:hystrix-servo-metrics-publisher:jar:1.1.2:compile
com.netflix.servo:servo-core:jar:0.4.27:compile
com.oracle:ojdbc5:jar:11.2.0.4:provided
com.oracle:ojdbc6:jar:11.2.0.3.0:compile
com.oracle:ucp:jar:11.2.0.1.0:compile
com.oracle.xdk:orai18n:jar:11.2.0.1.0:compile
com.oracle.xdk:xdb:jar:11.2.0.1.0:compile
com.renderx:xep_server:jar:3.7.1:compile
com.renderx:xt:jar:1.0.0:compile
com.sun.xml.bind:jaxb-impl:jar:2.1.3:compile
com.sun.xml.wsit:wsit-rt:jar:1.1:compile
com.sun.xml.wsit:xws-security:jar:1.3.1:compile
com.thetransactioncompany:cors-filter:jar:2.1.1:compile
com.thetransactioncompany:java-property-utils:jar:1.9.1:compile
com.thoughtworks.xstream:xstream:jar:1.4.2:compile
com.twilio.sdk:twilio-java-sdk:jar:6.3.0:compile
com.wordnik:swagger-annotations:jar:1.3.11:compile
com.yammer.metrics:metrics-core:jar:2.2.0:compile
com.yammer.metrics:metrics-graphite:jar:2.2.0:compile
com.yammer.metrics:metrics-logback:jar:2.2.0:compile
commons-beanutils:commons-beanutils-core:jar:1.8.3:compile
commons-beanutils:commons-beanutils:jar:1.8.3:compile
commons-codec:commons-codec:jar:1.6:compile
commons-collections:commons-collections:jar:3.2.1:compile
commons-configuration:commons-configuration:jar:1.8:runtime
commons-dbcp:commons-dbcp:jar:1.4:compile
commons-digester:commons-digester:jar:1.8.1:compile
commons-io:commons-io:jar:2.4:compile
commons-lang:commons-lang:jar:2.6:compile
commons-pool:commons-pool:jar:1.5.4:compile
commons-validator:commons-validator:jar:1.6:compile
dom4j:dom4j:jar:1.6.1:compile
io.jsonwebtoken:jjwt:jar:0.4:compile
io.reactivex:rxjava:jar:1.0.13:compile
javassist:javassist:jar:3.12.0.GA:compile
javax.activation:activation:jar:1.1:compile
javax.inject:javax.inject:jar:1:compile
javax.jms:jms:jar:1.1:compile
javax.servlet:servlet-api:jar:2.5:provided
javax.transaction:jta:jar:1.1:compile
javax.validation:validation-api:jar:1.0.0.GA:compile
javax.xml.bind:jaxb-api:jar:2.1:compile
javax.xml.stream:stax-api:jar:1.0-2:compile
joda-time:joda-time-hibernate:jar:1.3:compile
joda-time:joda-time:jar:2.2:compile
junit:junit:jar:4.11:test
net.jpountz.lz4:lz4:jar:1.2.0:compile
net.logstash.logback:logstash-logback-encoder:jar:1.2:compile
net.minidev:json-smart:jar:1.2:compile
net.sf.dozer:dozer:jar:5.4.0:compile
net.sf.ehcache:ehcache-core:jar:2.4.5:compile
org.apache.commons:commons-lang3:jar:3.1:compile
org.apache.httpcomponents:httpclient-cache:jar:4.2.1:compile
org.apache.httpcomponents:httpclient:jar:4.3.4:compile
org.apache.httpcomponents:httpcore:jar:4.3.2:compile
org.apache.ibatis:ibatis-sqlmap:jar:2.3.4.726:compile
org.apache.kafka:kafka-clients:jar:0.9.0.0:compile
org.apache.maven.surefire:surefire-api:jar:2.14.1:test
org.apache.maven.surefire:surefire-junit4:jar:2.14.1:test
org.apache.santuario:xmlsec:jar:1.4.2:compile
org.apache.velocity:velocity:jar:1.6.2:compile
org.apache.ws.security:wss4j:jar:1.5.5:compile
org.apache.xmlbeans:xmlbeans:jar:2.5.0:compile
org.aspectj:aspectjweaver:jar:1.8.2:compile
org.assertj:assertj-core:jar:2.5.0:compile
org.codehaus.jackson:jackson-core-asl:jar:1.9.3:compile
org.codehaus.jackson:jackson-mapper-asl:jar:1.9.3:compile
org.easymock:easymock:jar:3.0:test
org.exparity:hamcrest-date:jar:1.1.0:compile
org.glassfish:javax.annotation:jar:10.0-b28:compile
org.hamcrest:hamcrest-all:jar:1.3:compile
org.hamcrest:hamcrest-core:jar:1.3:compile
org.hamcrest:hamcrest-library:jar:1.3:test
org.hibernate:hibernate-commons-annotations:jar:3.2.0.Final:compile
org.hibernate:hibernate-core:jar:3.6.6.Final:compile
org.hibernate:hibernate-entitymanager:jar:3.6.6.Final:compile
org.hibernate:hibernate-validator:jar:4.0.2.GA:compile
org.hibernate.javax.persistence:hibernate-jpa-2.0-api:jar:1.0.1.Final:compile
org.jasig.cas:cas-client-core:jar:3.1.10:compile
org.jasig.cas.client:cas-client-core:jar:3.1.11:compile
org.javassist:javassist:jar:3.16.1-GA:compile
org.jaxen:com.springsource.org.jaxen:jar:1.1.1:compile
org.jibx:jibx-run:jar:1.2.4.5:compile
org.mockito:mockito-core:jar:1.9.5:test
org.mybatis:mybatis-spring:jar:1.1.0:compile
org.mybatis:mybatis:jar:3.1.0:compile
org.objenesis:objenesis:jar:1.2:compile
org.reflections:reflections:jar:0.9.9-RC1:compile
org.slf4j:jcl-over-slf4j:jar:1.7.5:compile
org.slf4j:jul-to-slf4j:jar:1.7.5:compile
org.slf4j:log4j-over-slf4j:jar:1.7.5:compile
org.slf4j:slf4j-api:jar:1.7.5:compile
org.springframework:spring-aop:jar:3.2.9.RELEASE:compile
org.springframework:spring-aspects:jar:3.2.9.RELEASE:compile
org.springframework:spring-beans:jar:3.2.9.RELEASE:compile
org.springframework:spring-context-support:jar:3.2.9.RELEASE:compile
org.springframework:spring-context:jar:3.2.9.RELEASE:compile
org.springframework:spring-core:jar:3.2.9.RELEASE:compile
org.springframework:spring-expression:jar:3.2.9.RELEASE:compile
org.springframework:spring-jdbc:jar:3.2.9.RELEASE:compile
org.springframework:spring-jms:jar:3.2.9.RELEASE:compile
org.springframework:spring-orm:jar:3.2.9.RELEASE:compile
org.springframework:spring-oxm:jar:3.2.9.RELEASE:compile
org.springframework:spring-test:jar:3.2.9.RELEASE:test
org.springframework:spring-tx:jar:3.2.9.RELEASE:compile
org.springframework:spring-web:jar:3.2.9.RELEASE:compile
org.springframework:spring-webmvc:jar:3.2.9.RELEASE:compile
org.springframework.data:spring-data-commons:jar:1.7.0.RELEASE:compile
org.springframework.data:spring-data-jpa:jar:1.5.0.RELEASE:compile
org.springframework.hateoas:spring-hateoas:jar:0.8.0.RELEASE:compile
org.springframework.retry:spring-retry:jar:1.0.3.RELEASE:compile
org.springframework.security:spring-security-cas-client:jar:3.0.7.RELEASE:compile
org.springframework.security:spring-security-config:jar:3.2.5.RELEASE:compile
org.springframework.security:spring-security-core:jar:3.2.5.RELEASE:compile
org.springframework.security:spring-security-web:jar:3.2.5.RELEASE:compile
org.springframework.ws:spring-ws-core:jar:2.1.4.RELEASE:compile
org.springframework.ws:spring-ws-security:jar:2.1.4.RELEASE:compile
org.springframework.ws:spring-ws-support:jar:2.1.4.RELEASE:compile
org.springframework.ws:spring-ws:jar:all:2.0.5.RELEASE:compile
org.springframework.ws:spring-xml:jar:2.1.4.RELEASE:compile
org.xerial.snappy:snappy-java:jar:1.1.1.7:compile
org.yaml:snakeyaml:jar:1.10:compile
oro:oro:jar:2.0.8:compile
wsdl4j:wsdl4j:jar:1.6.1:compile
xalan:serializer:jar:2.7.1:compile
xalan:xalan:jar:2.7.1:compile
xerces:xercesImpl:jar:2.9.1:compile
xml-apis:xml-apis:jar:1.3.04:compile
xmlpull:xmlpull:jar:1.1.3.1:compile
xpp3:xpp3_min:jar:1.1.4c:compile
xpp3:xpp3:jar:1.1.3.4.O:compile
```

Cheers

**Comment by mjedynak** [comment]
fixed in 1.3.8
